apollo-router/tests/integration/subgraph_mocks_plugin/mod.rs

```diff
@@ -0,0 +1,270 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use apollo_compiler::ExecutableDocument;
+use apollo_compiler::Schema;
+use apollo_compiler::ast::OperationType;
+use apollo_compiler::request::coerce_variable_values;
+use apollo_compiler::response::GraphQLError;
+use apollo_compiler::response::JsonMap;
+use apollo_compiler::response::JsonValue;
+use apollo_compiler::validation::Valid;
+use apollo_router::_private::new_subgraph_response;
+use apollo_router::graphql;
+use apollo_router::plugin::Plugin;
+use apollo_router::plugin::PluginInit;
+use apollo_router::services::subgraph;
+use tower::BoxError;
+use tower::ServiceExt;
+
+use self::execution::resolver::ResolvedValue;
+
+mod execution;
+
+apollo_router::register_plugin!("apollo", "subgraph_mocks", SubgraphMocksPlugin);
+
+/// Example of inline `router.yaml` config:
+///
+/// ```yaml
+/// subgraph_mocks:
+///   subgraph1_name:
+///     query:
+///       rootField:
+///         subField: "value"
+///     entities:
+///       - __typename: Something
+///         id: 4
+///         field: [42, 7]
+/// ```
+///
+/// Example with mocked data in separate `mocked_subgraph_data.yaml` file:
+/// TODO: what’s the base path for this relative filename?
+///
+/// ```yaml
+/// subgraph_mocks: ${file.mocked_subgraph_data.yaml}
+/// ```
+type Config = HashMap<String, Arc<SubgraphConfig>>;
```

Comment:
I was going to raise this during the presentation you were giving, but we ran out of time. Could you write either a confluence doc or a dev doc in the repo on how you'd like people to use this new and improved way of mocking subgraphs? I think if a month from now I'll go to write a test like this, I am going to forget which files to look for examples in 😅